### PR TITLE
Add the rabbitmq role

### PR DESCRIPTION
**Pull request by mprahl**

### CODE REVIEW

**Comment by mprahl** [comment]
@lcarva and @yashvardhannanavati could you please review?

File: README.md [hunk 1]
```diff
@@ -1,2 +1,79 @@
...
+  and to use for AMQPS and HTTPS on the management interface. If this is not set, AMQP and HTTP
+  will be used instead. This is not defined by default.
+* `rabbitmq_users` - the list of RabbitMQ users to create. This uses the same syntax as the
+  [rabbitmq_user_module](https://docs.ansible.com/ansible/latest/modules/rabbitmq_user_module.html).
+  This defaults to `[]`.
+
+Example Playbook
+----------------
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
```
Comment (lcarva) [inline README.md]: Maybe remove this line?

File: tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  rabbitmq_user:
+    user: guest
+    state: absent
+  when: rabbitmq_remove_guest_user | bool
+  tags:
+    - rabbitmq
+
+- name: Add RabbitMQ users
+  rabbitmq_user:
+    configure_priv: "{{ item.configure_priv | default(omit) }}"
```
Comment (lcarva) [inline tasks/main.yml]: Is there a more flexible way of doing this?  Maybe `rabbitmq_user: "{{ item }}"` ?

File: tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    password: "{{ item.password | default(omit) }}"
+    permissions: "{{ item.permissions | default(omit) }}"
+    read_priv: "{{ item.read_priv | default(omit) }}"
+    tags: "{{ item.tags | default(omit) }}"
+    update_password: "{{ item.update_password | default(omit) }}"
+    user: "{{ item.user }}"
+    vhost: "{{ item.vhost | default(omit) }}"
+    write_priv: "{{ item.write_priv | default(omit) }}"
+  with_items: "{{ rabbitmq_users }}"
+  no_log: yes
```
Comment (lcarva) [inline tasks/main.yml:164]: Let's add a comment as to why `no_log` is set to `true`.

File: tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    package: "{{ item }}"
+  with_items:
+    - rabbitmq-server
+    - erlang
+  when: rabbitmq_version_lock | bool
+  tags:
+    - rabbitmq
+
+- name: Copy over the SSL key
+  copy:
```
Comment (lcarva) [inline tasks/main.yml:75]: This should have `no_log: true`, right?

File: tasks/main.yml (hunk 2 above)
Comment (mprahl) [reply tasks/main.yml]: I didn't know you could do that. That's much better!

File: tasks/main.yml (hunk 4 above)
Comment (mprahl) [reply tasks/main.yml:75]: I didn't think an error would expose the file contents, but I set no_log to be safe.

File: tasks/main.yml (hunk 2 above)
Comment (lcarva) [reply tasks/main.yml]: Hey, I don't know if it'll work! Try it out first :)

**Review by lcarva — APPROVED** [review]

**Pull request merged**